### PR TITLE
Add Metric Relationship

### PR DESCRIPTION
**Pull request by dan-tang-ssd**
This PR is submitted for below changes:
1. Remove filament resources for topics, sub-dimensions, farming systems
2. Add relation manager for metric and other entities
3. Hide topics, sub-dimensions, farming systems in Metric filament resource

---

I have not removed database tables topics, sub_dimensions, farming_systems yet. Not quite sure whether topics table is required in Metric edit page. We can remove these tables in later stage.

### CODE REVIEW

**Comment by dan-tang-ssd** [comment]
TODO: Add filament resource for Themes

File: app/Filament/Resources/MetricResource.php [hunk 1]
```diff
@@ -143,15 +143,17 @@ public static function form(Form $form): Form
 
                             ]),
 
-                        Tab::make('Topics and Dimensions')
+                        Tab::make('Dimensions')
```
Comment (dave-mills) [inline app/Filament/Resources/MetricResource.php]: Given we're not using topics at all (It seems clear that they were replaced by "themes" in the ICRAF team's thinking), I think we definitely can remove topics from here and entirely clean them from the database.

File: app/Filament/Resources/MetricResource.php [hunk 2]
```diff
@@ -175,34 +177,35 @@ public static function form(Form $form): Form
...
+                                //             ->relationship('dimension', 'name'),
+                                //         Select::make('parent_id')
+                                //             ->relationship('parent', 'name')
+                                //             ->label('Is this is a descendant of another sub-dimension?'),
+                                //         TextInput::make('name'),
+                                //         Textarea::make('notes'),
+                                //     ])
+                                //     ->multiple()
+                                //     ->createOptionAction(fn (Action $action): Action => $action->tooltip('Create new'))
+                                //     ->suffixAction(self::makeDiscussionPointAction()),
```
Comment (dave-mills) [inline app/Filament/Resources/MetricResource.php]: Let's completely delete this - it's in the git history if we ever do need to go back.

**Review by dave-mills — COMMENTED** [review]
Nice! Good quick changes. Thanks. 

A few comments: 

- I think we can completely remove Topic and FarmingSystem from the whole platform. I'd rather have a cleaner view of what we do have, without keeping the parts from the original ideas that we aren't intending to use in the final system. 

- In the relation managers, could you also show whether the relationship needs review? And then let the user change that in the editing. 

- I'd check the metric users relation manager - I can't attach new metrics there, because there isn't a `metricUsers()` relationship on `Metric::class`. (That's one of the relationships where there are actually 3 ways a `MetricUser` can be linked to a Metric). I'd suggest for now adding a generic relationship so we can see *all* the linked entries, regardless of what `type` is in the pivot table.

**Comment by dan-tang-ssd** [comment]
Hi Dave, Thank you for your feedback.

>     * I think we can completely remove Topic and FarmingSystem from the whole platform. I'd rather have a cleaner view of what we do have, without keeping the parts from the original ideas that we aren't intending to use in the final system.

I have questions about removing Topic...

In front end, dimension checks for selected topics. If I completely remove Topic instead of hiding it, dimension checkbox list options() modifiers will not work.

Questions:
1. Should we keep to show dimensions in front-end?
2. Should we remove the hierarchy between topic and dimension?


image

image

**Comment by dave-mills** [comment]
Yes - removing topics implies removing the relationship between Topic and Dimension. 18 out of the 4611 dimension entries in the database have a topic linked, so we are not losing much. We're losing even less given we're simply not using topics any more (as they have been replaced by themes).

Removing topics also implies modifying the UI for dimension creation / editing and metric creation / editing. I suggest that, for viewing / editing metrics, we entirely remove the "topics + dimensions" tab (as we now don't have topics or sub-dimensions), and add a dimensions relation manager so it's more in line with the other linked entity types.

**Review by dave-mills — APPROVED** [review]
Thanks for this - I think this is good to merge. There are a few things I think we can improve with the UI for managing these entries, but that can be a new PR for clarity and so we're not blocking the ability to publish this whole piece of work to the staging site.

**Pull request merged**